distributed/client.py

```diff
@@ -2096,9 +2114,16 @@ def compute(self, collections, sync=False, optimize_graph=True,
         if resources:
             resources = self.expand_resources(resources)
 
+        if retries:
+            # Each task unit may potentially fail, allow retrying all of them
+            retries = {name: retries for name in itertools.chain(dsk, dsk2)}
+        else:
+            retries = None
```

Comment:
This breaks a little bit from the convention we have for `resources`, where users can specify parts of their computation that have different resource constraints.

See this test for compute

```python
@gen_cluster(client=True, ncores=[('127.0.0.1', 1, {'resources': {'A': 1}}),
                                  ('127.0.0.1', 1, {'resources': {'B': 1}})])
def test_compute(c, s, a, b):
    x = delayed(inc)(1)
    y = delayed(inc)(x)

    yy = c.compute(y, resources={x: {'A': 1}, y: {'B': 1}})
    yield wait(yy)

    assert b.data
```

Arguably we should publish this convention in docs more prominently.  I suspect that the current `expand_resources` function already has the logic you would need here.  (that function should probably be renamed and made private).

Reply:
Ok, I improved `retries` handling to allow both an int or a dict of per-task specifications.